Cstyle linting for istg lu.c

diegous opened the pull request:
This file is very large and had many things to fix, so I grouped each type of fix in a different commit. 

I haven't yet fixed _"line > 80 characters"_ because I thought that it would make this PR too big and with too many things to check that could break.

Please, let me know what you guys think.

Fixes: openebs/openebs#2051

diegous committed:
Removed trailing whitespaces and last blank line in file

diegous committed:
Fixed: 'unparenthesized return expression'

diegous committed:
Fixed: '#define followed by space instead of tab'

diegous committed:
Fixed: 'missing space between keyword and paren'

diegous committed:
Fixed: 'improper first line of block comment'

diegous committed:
Fixed: comma or semicolon followed by non-blank

diegous committed:
Fixed: missing blank after start comment

diegous committed:
Fixed: unparenthesized sizeof expression

diegous committed:
Fixed: whitespace (before rigth|after left) paren

diegous committed:
Fixed: indent by spaces instead of tabs

diegous committed:
Fixed: improper boolean continuation

diegous committed:
Fixed: spaces instead of tabs

diegous committed:
Fixed: missing space around

diegous committed:
Fixed: missing space between type name and *

diegous committed:
Fixed: missing space (before left|after right) brace

diegous committed:
Fixed: left brace starting a line

diegous committed:
Fixed: else and right brace should be on same line

diegous committed:
Fixed: extra space between function name and left paren

diegous committed:
Fixed: more than one space around boolean operator

diegous committed:
Fixed: continuation line not indented by 4 spaces

diegous committed:
Fixed: unneeded return at end of function

diegous committed:
Fixed: missing blank (after open|before close) comment

diegous committed:
Fixed: return type of function not on separate line

payes committed:
Merge branch 'replication' into cstyle-linting-for-istg_lu.c

utkarshmani1997 commented on src/istgt_lu.c:
```diff
@@ -1007,11 +1006,11 @@ istgt_lu_parse_media_size(const char *file, const char *size, int *flags)
 		if (msize < ISTGT_LU_MEDIA_SIZE_MIN) {
 			ISTGT_ERRLOG("media size too small\n");
-			return 0ULL;
+			return (0ULL); // Shuld this be NULL?
```
it's used as `unsigned long long zero` . kindly remove the comment

utkarshmani1997 commented on src/istgt_lu.c:
```diff
@@ -2827,19 +2826,19 @@ istgt_lu_update_unit(ISTGT_LU_Ptr lu, CF_SECTION *sp)
+	// MTX_LOCK(&lu->istgt->mutex);
+	// lu->istgt->nlogical_unit++;
+	// lu->istgt->logical_unit[lu->num] = lu;
+	// MTX_UNLOCK(&lu->istgt->mutex);
```
shouldn't multiline comments be in `/*        */` this form ?

utkarshmani1997 commented on src/istgt_lu.c:
```diff
@@ -3130,34 +3129,34 @@ istgt_lu_match_all(CF_SECTION *sp, CONFIG *config_old)
+// 	istgt_copy_cf_item(sp, sp_old);
+// 	return (0);
+// }
 
```
same comment as earlier

utkarshmani1997 commented on src/istgt_lu.c:
```diff
@@ -3237,13 +3236,13 @@ istgt_lu_reload_update(ISTGT_Ptr istgt)
 							/* ignore error */
-						//}
+						// }
 						MTX_UNLOCK(&lu->mutex);
```
same comment as earlier

utkarshmani1997 commented on src/istgt_lu.c:
```diff
@@ -3939,18 +3940,16 @@ istgt_lu_create_task(CONN_Ptr conn, ISTGT_LU_CMD_Ptr lu_cmd, int lun, ISTGT_LU_D
+	// 	lu_task->lba != lu_task->lu_cmd.lba ||
+	// 	lu_task->lblen != lu_task->lu_cmd.lblen) {
+	// 	log/fix
+	// }
```
same comment as earlier

utkarshmani1997 reviewed (changes requested):
@diegous thanks for the fixes, this look good to me. Have provided the comments regarding multiline comments. Correct me if i'm wrong.

diegous committed:
Changed single line comments for block comments

diegous commented on src/istgt_lu.c:
```diff
@@ -3670,20 +3678,20 @@ istgt_lu_create_task(CONN_Ptr conn, ISTGT_LU_CMD_Ptr lu_cmd, int lun, ISTGT_LU_D
-	alloc_len = ISCSI_ALIGN(sizeof *lu_task);
-	alloc_len += ISCSI_ALIGN(sizeof *lu_task->lu_cmd.pdu);
+	alloc_len = ISCSI_ALIGN(sizeof (*lu_task));
+	alloc_len += ISCSI_ALIGN(sizeof *lu_task->lu_cmd.pdu); // I'm not sure if this is 'sizeof (*lu_task)->lu_cmd.pdu' or 'sizeof (*lu_task->lu_cmd.pdu)'
```
I wasn't sure how to solve this

vishnuitta replied:
you can make it as sizeof (struct iscsi_pdu_t)

vishnuitta committed:
Merge branch 'replication' into cstyle-linting-for-istg_lu.c

vishnuitta reviewed (approved):
changes are good.
Other cstyle error 'line > 80 characters' can be done in a different PR

vishnuitta committed:
Merge branch 'replication' into cstyle-linting-for-istg_lu.c

vishnuitta committed:
Merge branch 'replication' into cstyle-linting-for-istg_lu.c

vishnuitta committed:
Merge branch 'replication' into cstyle-linting-for-istg_lu.c

diegous committed:
Solved comment

diegous committed:
Merge branch 'cstyle-linting-for-istg_lu.c' of github.com:diegous/istgt into cstyle-linting-for-istg_lu.c

diegous committed:
Corrected indentation of preprocessor statements

diegous commented:
@utkarshmani1997 I've corrected the last of the comments, and fixed a couple of lines added when the 'replication' branch was updated. If everything looks good enough to merge we should probably squash all the commits before.
